libs/crypto/merkle-impl/src/main/kotlin/net/corda/crypto/merkle/impl/MerkleProofImpl.kt

```diff
@@ -80,47 +85,97 @@ class MerkleProofImpl(
         }
         var hashIndex = 0
         val sortedLeaves = leaves.sortedBy { it.index }
-        // work out nodeHashes, which is a map from leaf index to digest
-        var nodeHashes = sortedLeaves.map { Pair(it.index, digest.leafHash(it.index, it.nonce, it.leafData)) }
-        var treeDepth = MerkleTreeImpl.treeDepth(treeSize)
-        var currentSize = treeSize
+        // work out nodeHashes, which is a list of node information for the current level we operate at
```

Comment:
```suggestion
        // work out nodeHashes, which is a map of index to hash for the current level we operate at
```
Or something like this?

Reply:
Actually it is a list. Amusingly, I made the same mistake and thought it was a map when first reding the code, since the initialiser for it includes the word `map` prominently which is why I added the type annotation on L89.

Reply:
You are right. Sorry for the noise... :)